packages/dds/tree/src/feature-libraries/chunked-forest/codec/nodeShape.ts

```diff
@@ -51,14 +51,17 @@ export class NodeShape extends Shape<EncodedChunkShape> implements NodeEncoder {
 		}
 
 		if (this.value === 0) {
-			const sessionSpaceCompressedId = cache.idCompressor.tryRecompress(
-				cursor.value as StableId,
-			);
-			const opSpaceCompressedId =
-				sessionSpaceCompressedId !== undefined
-					? cache.idCompressor.normalizeToOpSpace(sessionSpaceCompressedId)
-					: cursor.value;
-			encodeValue(opSpaceCompressedId, this.value, outputBuffer);
+			assert(typeof cursor.value === "string", "identifier must be type string");
+			if (!isStableId(cursor.value)) {
```

Comment:
nit: for cases where there is one "success" path and multiple "fallback" paths, either structure it like this:
```ts
if (conditionA && conditionB) {
	// Success
} else {
	// Failure
}
```
or, in a helper function with returns like this:
```ts
{
	if (conditionA) {
		if (conditionB) {
			return success;
		}
	}
	return failure;
}
```

The current code is structured like:

```ts
if (!conditionA) {
	// Failure
} else {
	if (conditionB) {
		// Success
	} else {
		// Failure
	}
}
```

- note how there are two identical "failure" cases - i.e. the cases are not "factored out" in the algebraic sense. There are times when it's fine to have multiple cases too, like this, which reduces nesting:

```ts
{
	if (!conditionA) {
		return false;
	}

	if (!conditionB) {
		return false;
	}

	return true;
}
```

But all the "failure" cases are at least still together.